src/datasource.ts

```diff
@@ -16,12 +16,33 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import { DataSourceInstanceSettings } from '@grafana/data';
-import { DataSourceWithBackend } from '@grafana/runtime';
+import { DataSourceInstanceSettings, ScopedVars } from '@grafana/data';
+import { DataSourceWithBackend, getTemplateSrv } from '@grafana/runtime';
 import { PixieDataSourceOptions, PixieDataQuery } from './types';
 
 export class DataSource extends DataSourceWithBackend<PixieDataQuery, PixieDataSourceOptions> {
   constructor(instanceSettings: DataSourceInstanceSettings<PixieDataSourceOptions>) {
     super(instanceSettings);
   }
+
+  applyTemplateVariables(query: PixieDataQuery, scopedVars: ScopedVars) {
+    let timeVars = [
+      ['$__from', '__time_from'],
+      ['$__to', '__time_to'],
+      ['$__interval', '__interval'],
+    ];
+
+    timeVars.forEach((replace) => {
+      query.pxlScript = query.pxlScript.replaceAll(replace[0], replace[1]);
```

Comment:
For readability, can do something like 
```suggestion
    for (const [changeFrom, changeTo] of timeVars) {
        query.pxlScript = query.pxlScript.replaceAll(replaceFrom, replaceTo);
    }
```